### PR TITLE
Version upgrades

### LINKED ISSUE #115: Dependency Upgrades
We are far behind the versions of major packages we use in the project. Need to upgrade them to latest

- [ ] fastapi
- [ ] pydantic
- [ ] chromadb
- [ ] langchain

Make code changes for each of these and test them

### PR DESCRIPTION
**Pull request by kavitharaju**
- Closes #115 
- The PR includes version upgrades in requirements.txt and corresponding code changes for
  - [x] fastapi 0.95.1 -> 0.104.1
  - [x] pydantic 1.10.7 -> 2.4.2
  - [x] chromadb 0.3.22 -> 0.4.15
  - [x] langchain 0.0.165 -> 0.0.331
- Tested the chat and login parts manually and data upload via existing tests

### CODE REVIEW

File: app/tests/conftest.py [hunk 1]
```diff
@@ -43,8 +43,9 @@ def fresh_db():
     try:
         yield {"dbPath": chroma_db_path, "collectionName": collection_name}
     finally:
-        if os.path.exists(chroma_db_path):
-            shutil.rmtree(chroma_db_path)
+        chroma_client = chromadb.PersistentClient(path='chromadb_store_test')
+        chroma_client.delete_collection(name=collection_name)
+
         db_conn = psycopg2.connect(
```
Comment (alejandroaquintero) [inline app/tests/conftest.py:49]: Outside of the scope of the pr but having a utility function to run the postgress commands would make it easier to read.

**Review by alejandroaquintero — APPROVED** [review]

File: app/core/vectordb/chroma.py [hunk 2]
```diff
@@ -95,26 +84,28 @@ def add_to_collection(self, docs: List[schema.Document], **kwargs) -> None:
                 {
                     "label": doc.label,
                     "media": ",".join(doc.media),
-                    "links": ",".join(doc.links),
+                    "links": ",".join(str(doc.links)),
```
Comment (woodwardmw) [inline app/core/vectordb/chroma.py:87]: I'm curious how you can have a string parameter to `join()`...

**Review by woodwardmw — APPROVED** [review]
Looks good to me. If it passes tests and you've tested it locally, that's probably fine.

File: app/core/vectordb/chroma.py (hunk 2 above)
Comment (kavitharaju) [reply app/core/vectordb/chroma.py:87]: Probably a pydantic suggested change. Shall check this

Comment (kavitharaju) [reply app/core/vectordb/chroma.py]: Made a change for this. Our existing tests seems to be working in both cases though.

**Pull request merged**